### PR TITLE
fontconfig 2.13.91

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **fontconfig/2.13.91**

this is a port of https://github.com/conan-community/conan-fontconfig

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fontconfig/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            tools.replace_in_file("Makefile", "po-conf test", "po-conf")
+        return self._autotools
+
+    def _patch_files(self):
+        def rename_if_exists(old, new):
+            if os.path.exists(old):
+                os.rename(old, new)
+        #  - fontconfig requires libtool version number, change it for the corresponding freetype one
+        tools.replace_in_file(os.path.join(self._source_subfolder, 'configure'), '21.0.15', '2.8.1')
+        rename_if_exists('freetype.pc', 'freetype2.pc')
```
Comment (madebr) [inline recipes/fontconfig/all/conanfile.py:59]: This isn't necessary anymore, I think.
https://github.com/conan-io/conan-center-index/blob/1f1323b3e3f483bfb5c619dc5c8b34c3ab1eb158/recipes/freetype/all/conanfile.py#L127

File: recipes/fontconfig/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        la = os.path.join(self.package_folder, "lib", "libfontconfig.la")
+        if os.path.isfile(la):
+            os.unlink(la)
```
Comment (madebr) [inline recipes/fontconfig/all/conanfile.py:73]: The libtool file will be there. Always.
```suggestion
        os.unlink(os.path.join(self.package_folder, "lib", "libfontconfig.la"))
```

File: recipes/fontconfig/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "var"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (madebr) [inline recipes/fontconfig/all/conanfile.py:80]: The output of a random run showed me that only one library was packaged.
```suggestion
        self.cpp_info.libs = ["fontconfig"]
```

File: recipes/fontconfig/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    generators = "cmake", "cmake_find_package"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        bin_path = os.path.join("bin", "example")
+        env = {"FONTCONFIG_PATH": os.path.join(self.deps_cpp_info["fontconfig"].rootpath, "etc", "fonts")}
```
Comment (madebr) [inline recipes/fontconfig/all/test_package/conanfile.py:15]: You're removing the `etc` folder in `package`, so is this working?
Should this folder be installed elsewhere?

Should this be moved to `package_info` as a `self.env_info.FONTCONFIG_PATH = ...`?

**Comment by uilianries** [comment]
@danimtb Seems like Jenkins is a bit unstable. I saw some "Unexpected Error" returns on last builds.

**Comment by ericLemanissier** [comment]
thanks ! There seem to be no binaries available yet:
```
> conan search -r all fontconfig
Existing package recipes:

Remote 'conan-center':
fontconfig/2.13.91@conan/stable
```
Is it too early, or is there something wrong ?

**Comment by ericLemanissier** [comment]
ping @jgsogo @danimtb I still cannot see any fontconfig binaries on CCI

**Comment by jgsogo** [comment]
Yes, there has been a problem in the main job. This package is adding to the tar file some symlinks that are outside the package. The target files are available for the PR (it is building the package), but in the master job those files are no longer there (no build, just download, check the package and upload).

We will disable this check in the main job until we can run it for the PRs too.

Sorry for the inconvenience

**Comment by ericLemanissier** [comment]
it has some symbolink links, but they are pointing inside the package folder. They seem to be absolute links though:
```
/home/eric/.conan/data/fontconfig/2.13.91/_/_/package/4fb4ea7ea368e5d0c3aab0da2460e23f629d5393/
├── bin
│   ├── etc
│   │   └── fonts
│   │       ├── conf.d
│   │       │   ├── 10-hinting-slight.conf -> /home/eric/.conan/data/fontconfig/2.13.91/_/_/package/4fb4ea7ea368e5d0c3aab0da2460e23f629d5393/bin/share/fontconfig/conf.avail/10-hinting-slight.conf
```
Would we need a tool to transform all absolute symbolink links in a package into relative links ?

**Comment by jgsogo** [comment]
This is the error:

```
ERROR: fontconfig/2.13.91:4b00a0d9586015d2a6974d175da5311704f0c54a: Upload package to 'centralRepo' failed: The file is a broken symlink, verify that you are packaging the needed destination files: '/tmp/c3ipr/bintray_folder/.conan/data/fontconfig/2.13.91/_/_/package/4b00a0d9586015d2a6974d175da5311704f0c54a/bin/etc/fonts/conf.d/60-latin.conf'.You can skip this check adjusting the 'general.skip_broken_symlinks_check' at the conan.conf file.
```

Probably not a tool, there are many issues in Conan about symlinks and we need to decide what to do with them first: https://github.com/conan-io/conan/issues/7093

For Conan Center probably it is ok to relativize those links (I'm not sure if Conan is already doing it when creating the tar file) and probably we can add a hook that fails if any symlink is pointing outside the package. Can you imagine any package that needs to add a symlink to something outside the package itself?

**Comment by ericLemanissier** [comment]
I honestly don't know what should be done regarding links. I've not directly faced this issue before yesterday. Regarding fontconfig I just made a PR deleting them: https://github.com/conan-io/conan-center-index/pull/1816

**Comment by jgsogo** [comment]
`fontconfig` is already in Conan Center: https://conan.io/center/fontconfig/2.13.91/  At least the `master` job won't fail after the PR is merged (now we can consider #1816)

Thanks for your patience

**Pull request merged**